core/plugins/org.fusesource.ide.foundation.core/src/org/fusesource/ide/foundation/core/util/CamelUtils.java

```diff
@@ -183,4 +191,58 @@ public boolean visit(IResource resource) throws CoreException {
 		}
 		return files;
 	}
+	
+	/**
+	 * retrieves the camel design editor
+	 * 
+	 * @return
+	 */
+	private static IEditorPart getDiagramEditor() {
```

Comment:
it seems strange to me that we didn't have this kind of code somewhere else. Maybe i'm wrong, Have you checked for it?